File: rxrecycler-adapter/src/main/java/com/minimize/android/rxrecycleradapter/SimpleViewHolder.java
package com.minimize.android.rxrecycleradapter;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class SimpleViewHolder<T, V extends ViewDataBinding> extends RecyclerView.ViewHolder {
    private V mViewDataBinding;

    public V getViewDataBinding() {
        return mViewDataBinding;
    }

    public T getItem() {
        return mItem;
    }

    private T mItem;

    protected void setItem(final T item) {
        mItem = item;
    }

    public SimpleViewHolder(final View itemView) {
        super(itemView);
        mViewDataBinding = DataBindingUtil.bind(itemView);
    }
}

File: rxrecycler-adapter/src/main/java/com/minimize/android/rxrecycleradapter/SimpleViewHolder.kt
package com.minimize.android.rxrecycleradapter

import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

open class SimpleViewHolder<T, out V : ViewDataBinding>(itemView: View) :
    RecyclerView.ViewHolder(itemView) {
    val viewDataBinding: V? = DataBindingUtil.bind(itemView)

    var item: T? = null
}